webrender_traits/src/display_list.rs

```diff
@@ -340,28 +338,45 @@ impl DisplayListBuilder {
         self.push_new_empty_item(SpecificDisplayItem::PopStackingContext);
     }
 
-    pub fn push_scroll_layer(&mut self,
-                             clip: ClipRegion,
-                             content_size: LayoutSize,
-                             scroll_root_id: Option<ServoScrollRootId>) {
+    pub fn define_clip(&mut self,
+                       clip: ClipRegion,
+                       content_size: LayoutSize,
+                       scroll_root_id: Option<ServoScrollRootId>)
+                       -> ScrollLayerId {
         let scroll_layer_id = self.next_scroll_layer_id;
         self.next_scroll_layer_id += 1;
 
-        let scroll_layer_id = ScrollLayerId::new(self.pipeline_id, scroll_layer_id);
-        let item = SpecificDisplayItem::PushScrollLayer(PushScrollLayerItem {
+        let id = ScrollLayerId::new(self.pipeline_id, scroll_layer_id);
+        let item = SpecificDisplayItem::Clip(ClipDisplayItem {
             content_size: content_size,
-            id: scroll_layer_id,
+            id: id,
+            parent_id: *self.clip_stack.last().unwrap(),
             scroll_root_id: scroll_root_id,
         });
 
         self.push_item(item, clip.main, clip);
-        self.scroll_layer_stack.push(scroll_layer_id);
+        id
+    }
+
+    pub fn push_scroll_layer(&mut self,
```

Comment:
That's right. The idea is that this is a transitional PR. We will land this, update all clients to use the new API and then remove the old API.